.github/workflows/test-agent.yml

```diff
@@ -0,0 +1,143 @@
+# Copyright 2020 New Relic Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: test-pull-request
+
+#
+# Controls when the action will run. 
+#
+on:
+  #
+  # Triggers the workflow on push or pull request events but only for the main branch
+  #
+  push:
+    branches: 
+      - main
+      - 'dev'
+  pull_request:
+    branches: 
+      - main
+      - 'dev'
+
+jobs:
+  daemon-unit-tests:
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_NAME: newrelic/nr-php-agent-builder
+      IMAGE_TAG: make-go
+      IMAGE_VERSION: v1
+    strategy:
+      matrix:
+        platform: [gnu, musl]
+        arch: [amd64, arm64]
+    steps:
+      - name: Checkout newrelic-php-agent code
+        uses: actions/checkout@v3
+        with:
+          path: php-agent
+      - name: Enable arm64 emulation
+        if: ${{ matrix.arch == 'arm64' }}
+        run: |
+          docker run --privileged --rm tonistiigi/binfmt --install arm64
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build daemon
+        run: >
+          docker run --rm --platform linux/${{matrix.arch}}
+          -v "${GITHUB_WORKSPACE}/php-agent":"/usr/local/src/newrelic-php-agent" 
+          $IMAGE_NAME:$IMAGE_TAG-${{ matrix.platform }}-$IMAGE_VERSION daemon
+      - name: Run daemon tests
+        run: >
+          docker run --rm --platform linux/${{matrix.arch}}
+          -v "${GITHUB_WORKSPACE}/php-agent":"/usr/local/src/newrelic-php-agent" 
+          $IMAGE_NAME:$IMAGE_TAG-${{ matrix.platform }}-$IMAGE_VERSION daemon_test
+  agent-unit-test:
+    runs-on: ubuntu-latest
+    env:
+      IMAGE_NAME: newrelic/nr-php-agent-builder
+      IMAGE_TAG: make-php
+      IMAGE_VERSION: v1
+    strategy:
+      matrix:
+        platform: [gnu, musl]
+        arch: [amd64, arm64]
+        php: ['5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1', '8.2']
+        exclude:
+          - arch: arm64
+            php: '5.5'
+          - arch: arm64
+            php: '5.6'
+          - arch: arm64
+            php: '7.0'
+          - arch: arm64
+            php: '7.1'
+          - arch: arm64
+            php: '7.2'
+          - arch: arm64
+            php: '7.3'
+          - arch: arm64
+            php: '7.4'
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+        with:
+          path: php-agent
+      - name: Enable arm64 emulation
+        run: |
+          docker run --privileged --rm tonistiigi/binfmt --install arm64
```

Comment:
Thanks for this good suggestion. This is best to be done by using [docker-setup-qemu](https://github.com/marketplace/actions/docker-setup-qemu) action, and providing a pinned version of `tonistiigi/binfmt` image as `image` input. It will be done in its own PR.